assets/javascripts/discourse/components/full-page-chat.js

```diff
@@ -46,10 +47,31 @@ export default Component.extend({
 
     this.appEvents.off("chat:refresh-channels", this, "refreshModel");
     window.removeEventListener("resize", this._calculateHeight, false);
+    document.removeEventListener("keydown", this._autoFocusChatComposer);
     document.body.classList.remove("has-full-page-chat");
     this.chat.set("fullScreenChatOpen", false);
   },
 
+  @bind
+  _autoFocusChatComposer(e) {
+    if (!e.target || e.target.classList.contains("chat-composer-input")) {
```

Comment:
Won't this auto-focus logic trigger when using the search dropdown or when typing into a modal?

Reply:
hmm yeah I'm curious how to do this safely 🤔

Reply:
I can't have a list of elements to avoid b/c that will break as things are added. But maybe a list of selectors where if the activeElement is in X array or selectors... then we trigger this? IDK

Reply:
You could check if the active element is an input/textarea/etc. but still, David's concern about global app shortcuts is valid